instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/SqsEndpointModel.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.apachecamel;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import org.apache.camel.Endpoint;
+
+public class SqsEndpointModel {
+
+  private final Endpoint endpoint;
+
+  public SqsEndpointModel(Endpoint endpoint) {
+    this.endpoint = endpoint;
+  }
+
+  private Object getSqsConfiguration()
+      throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+    Method method = endpoint.getClass().getMethod("getConfiguration");
```

Comment:
Consider using `MethodHandle`s instead, as they're much faster than reflection (see `CouchbaseQueryNormalizer` for examples)

Reply:
Done :) Tho in the configuration case I had to use reflection as SqsConfiguration (rtype) can also be unavailable in the runtime.